src/viam/ur/module/ur_arm.cpp

```diff
@@ -422,9 +422,13 @@ ProtoStruct URArm::do_command(const ProtoStruct& command) {
     constexpr char k_get_tcp_forces_base_key[] = "get_tcp_forces_base";
     constexpr char k_get_tcp_forces_tool_key[] = "get_tcp_forces_tool";
     constexpr char k_clear_pstop[] = "clear_pstop";
+    constexpr char k_arm_ready[] = "arm_ready";
+    constexpr char k_arm_status[] = "arm_status";
 
     // Cache TCP state to ensure atomic read of pose and forces from same timestamp
     std::optional<decltype(current_state_->read_tcp_state_snapshot())> cached_tcp_state;
+    // Cache arm state description
+    auto description = current_state_->describe();
```

Comment:
Agree on getting the value outside the loop, but you could use the same `optional` trick I do above to avoid getting it if you don't or won't need it.